span: Add strict mode

miretskiy opened the pull request:
Two commits to help debug #116661 

1.
Log "slow span" messages both from the coordinator and from the aggregators.

2.
span: Add strict mode

Add strict mode for span frontier implementations that
requires the forwarded span to be a sub-span of the spans
tracked by the frontier.

The functionality is opt-in, and is enabled via
`COCKROACH_SPAN_FRONTIER_STRICT_MODE_ENABLED` env var.

Enroll cdc roachtests to use strict mode.

Informs #116661

Release Note: None

Commit:
changefeedcc: Expand slow span logging

Log "slow span" messages both from the coordinator
and from the aggregators.

Informs #116661

Release Note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/117097)
<!-- Reviewable:end -->

Commit:
span: Add strict mode

Add strict mode for span frontier implementations that
requires the forwarded span to be a sub-span of the spans
tracked by the frontier.

The functionality is opt-in, and is enabled via
`COCKROACH_SPAN_FRONTIER_STRICT_MODE_ENABLED` env var.

Enroll cdc roachtests to use strict mode.

Informs #116661

Release note: None
